projects/risk-assessment-scores-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/Event.kt

```diff
@@ -0,0 +1,135 @@
+import jakarta.persistence.OneToOne
+import jakarta.persistence.Table
+import org.hibernate.annotations.Immutable
+import org.hibernate.annotations.Where
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Lock
+import org.springframework.data.jpa.repository.Query
+import java.time.LocalDate
+import java.time.ZonedDateTime
+
+@Immutable
+@Entity
+class Event(
+    @Id
+    @Column(name = "event_id", nullable = false)
+    val id: Long,
+
+    @Column(name = "event_number", nullable = false)
+    val number: String,
+
+    @ManyToOne
+    @JoinColumn(name = "offender_id", nullable = false)
+    val person: Person,
+
+    @OneToOne(mappedBy = "event")
+    val disposal: Disposal? = null,
+
+    @OneToOne(mappedBy = "event")
+    val mainOffence: MainOffence? = null,
+
+    @Column(name = "active_flag", columnDefinition = "number", nullable = false)
+    val active: Boolean,
+
+    @Column(name = "soft_deleted", columnDefinition = "number")
+    val softDeleted: Boolean
+)
+
+@Immutable
+@Entity
+class Disposal(
+    @Id
+    @Column(name = "disposal_id")
+    val id: Long,
+
+    @OneToOne
+    @JoinColumn(name = "event_id", updatable = false)
+    val event: Event,
+
+    @ManyToOne
+    @JoinColumn(name = "disposal_type_id", updatable = false)
+    val disposalType: DisposalType,
+
+    @Column(name = "disposal_date", nullable = false)
+    val disposalDate: ZonedDateTime,
+
+    @Column(name = "active_flag", updatable = false, columnDefinition = "NUMBER")
+    val active: Boolean = true,
+
+    @Column(updatable = false, columnDefinition = "NUMBER")
+    val softDeleted: Boolean = false
+)
+
+@Immutable
+@Entity
+@Table(name = "r_disposal_type")
+class DisposalType(
+    @Id
+    @Column(name = "disposal_type_id")
+    val id: Long,
+
+    @Column(name = "description")
+    val description: String
+)
+
+@Immutable
+@Entity
+@Table(name = "main_offence")
+@Where(clause = "soft_deleted = 0")
+class MainOffence(
+    @Id
+    @Column(name = "main_offence_id")
+    val id: Long,
+
+    @OneToOne
+    @JoinColumn(name = "event_id")
+    val event: Event,
+
+    @ManyToOne
+    @JoinColumn(name = "offence_id")
+    val offence: Offence,
+
+    @Column(name = "offence_date")
+    val date: LocalDate,
+
+    @Column(name = "offence_count")
+    val offenceCount: Int,
+
+    @Column(updatable = false, columnDefinition = "NUMBER")
+    val softDeleted: Boolean = false
+)
+
+@Immutable
+@Entity
+@Table(name = "r_offence")
+class Offence(
+    @Id
+    @Column(name = "offence_id")
+    val id: Long,
+    val mainCategoryDescription: String
+)
+
+interface EventRepository : JpaRepository<Event, Long> {
+
+    @Query(
+        """
+        select e from Event e 
+        where e.person.crn = :crn
+        and e.number = :eventNumber
+        and e.softDeleted = false
+    """
+    )
+    fun findByCrn(crn: String, eventNumber: String): Event?
+
+    @Lock(LockModeType.PESSIMISTIC_READ)
```

Comment:
Save what on the event? I'm not making any changes to the event - plus it would mean adding the event to Access.yml AND fleshing out all the event fields. Seems less simple to me. Having said that I'm not sure the @Lock works as I'm trying to put in a test for it.

Reply:
You don't have to make any changes to the event, JPA will bump the row version automatically.  So you could create an Event entity with just the id, version, and created/updated fields.  Maybe it's just me - but pessimistic locking on the event table feels dangerous.